tests/e2e/factories/src/models/model.ts

```diff
@@ -0,0 +1,25 @@
+export abstract class Model {
+	private id: number | null = null;
+
+	/**
+	 * Fetches the model's ID.
+	 *
+	 * @return {number|null} The ID of the model or null if it has not yet been created.
+	 */
+	public get ID(): number | null {
+		return this.id;
+	}
+
+	/**
+	 * A hook to be called when the model is created, so that server-generated data can be stored.
+	 *
+	 * @param {number} id
+	 */
+	public onCreated( id: number ): void {
+		if ( null !== this.id ) {
+			throw new Error( 'The model has already been created!' );
```

Comment:
should we throw errors if the module is already created? why not skip or update silently?

Reply:
I'm generally a fan of throwing errors as explicitly and as early as possible. I find that programming defensively prevents people from using things in a way that we didn't intend (because they almost always do!)